### PR TITLE
abbreviate long token names

### PR DESCRIPTION
**Pull request by adub671**
## Description

Created an abbreviator function and applied to symbols in both Asset Supply Row & Asset Borrow Row. Popup Tooltip appears upon hover of abbreviated symbols

## Type of change

- [x] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## High-level change(s) description - from the user's perspective

Pop up tool tip with full symbol name appears for characters longer than 9 characters for better formatting 

## Related Issue(s)

Fixes

### CODE REVIEW

**Comment by adub671** [comment]
image
image

first screenshot is from PR. second is from my UI. I am only seeing the plugin symbol, I am not sure why this discrepancy exists

File: packages/ui/utils/abbreviator.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Text } from '@chakra-ui/react';
+import { ReactNode } from 'react';
+
+import { PopoverTooltip } from '@ui/components/shared/PopoverTooltip';
+
+export function abbreviator(str: string | undefined, len: number): string | undefined | ReactNode {
```
Comment (peetzweg) [inline packages/ui/utils/abbreviator.tsx:6]:
```suggestion
export function abbreviator(str?: string, len: number): string | undefined | ReactNode {
```

Comment (peetzweg) [reply packages/ui/utils/abbreviator.tsx:6]: I think if it's either the value or undefined using the `?` optional operator is a bit cleaner in my opinion. But we could keep it as is. ✌️

File: packages/ui/utils/abbreviator.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  if (str === undefined) return undefined;
+  return str?.length <= len ? (
+    str
+  ) : (
+    <PopoverTooltip placement="top-start" body={str}>
```
Comment (peetzweg) [inline packages/ui/utils/abbreviator.tsx:11]: Interesting to insert a popover around the abbreviated text to show the original string.
I would be a bit cautious if it really works as expected in every part of the app, as I don't know the inner workings of the Popover at all.

**Comment by peetzweg** [comment]
For reference there is also a CSS property to handle overflowing text:

https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow

**Comment by peetzweg** [comment]
> first screenshot is from PR. second is from my UI. I am only seeing the plugin symbol, I am not sure why this discrepancy exists

@adub671 so there is still a bug left? Maybe it's due to using the Popover? Maybe try it without the Popover. Using the primitive html `<span title="long text">abbreviated text</span>` should work for sure.

File: packages/ui/components/pages/Fuse/FusePoolPage/SupplyList/AssetSupplyRow.tsx [hunk 3]
```diff
@@ -359,7 +360,9 @@ export const AssetSupplyRow = ({
                   Number(utils.formatUnits(asset.supplyBalance, asset.underlyingDecimals)) <
                     UP_LIMIT &&
                   '+'}{' '}
-                {tokenData?.extraData?.shortName ?? tokenData?.symbol ?? asset.underlyingSymbol}
+                {abbreviator(tokenData?.extraData?.shortName, 9) ??
```
Comment (adub671) [inline packages/ui/components/pages/Fuse/FusePoolPage/SupplyList/AssetSupplyRow.tsx:363]: @peetzweg are we still using `tokenData.extraData.shortname` 
I noticed this only exists in `AssetSupplyRow.tsx` and is omitted in `AssetBorrowRow.tsx`

Let me know if I should remove it or add it to borrow row

**Comment by adub671** [comment]
features outdated due to new interface for supply and borrow row

**Pull request closed without merging**